subprojects/language-native/src/main/java/org/gradle/language/cpp/CppPlatform.java

```diff
@@ -26,4 +28,11 @@
  */
 @Incubating
 public interface CppPlatform extends NativePlatform {
+    /**
+     * The operating system family being targeted.
+     *
+     * @since 4.7
+     */
+    @Nested
+    OperatingSystemFamily getOperatingSystemFamily();
```

Comment:
I wonder if there might be some value in pulling this up into a super-interface, so we can treat `CppPlatform` and `SwiftPlatform` as a certain kind of `NativePlatform`?  Not sure there is, just asking the question...

Reply:
Yeah, I originally had this in `NativePlatform`, which I think is where I'd rather have this, but that type has mutability problems and already has an operating system. 

We should probably do more detangling/removal? of NativePlatform in https://github.com/gradle/gradle-native/issues/596

Reply:
We shouldn't have multiple methods for querying the operating system details of a C++/Swift/whatever platform. So, instead of adding a query method for the family, we should change this type so that it does not extend the software model's `NativePlatform`.